src/output-json-ike.c

```diff
@@ -0,0 +1,189 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Frank Honza <frank.honza@dcso.de>
+ *
+ * Implement JSON/eve logging app-layer IKE.
+ */
+
+#include "suricata-common.h"
+#include "debug.h"
+#include "detect.h"
+#include "pkt-var.h"
+#include "conf.h"
+
+#include "threads.h"
+#include "threadvars.h"
+#include "tm-threads.h"
+
+#include "util-unittest.h"
+#include "util-buffer.h"
+#include "util-debug.h"
+#include "util-byte.h"
+
+#include "output.h"
+#include "output-json.h"
+
+#include "app-layer.h"
+#include "app-layer-parser.h"
+
+#include "app-layer-ike.h"
+#include "output-json-ike.h"
+#include "rust.h"
+
+#define LOG_IKE_DEFAULT  0
+#define LOG_IKE_EXTENDED (1 << 0)
+
+typedef struct LogIKEFileCtx_ {
+    LogFileCtx *file_ctx;
+    uint32_t flags;
+} LogIKEFileCtx;
+
+typedef struct LogIKELogThread_ {
+    LogIKEFileCtx *ikelog_ctx;
+    uint32_t count;
+    MemBuffer *buffer;
```

Comment:
To support eve multi-threaded output, you'll need to add the `file_ctx` to this structure (see example from `output-json-ikev2.c`).

Reply:
This is a new feature, right ?